packages/next-codemod/transforms/next-lint-to-eslint-cli.ts

```diff
@@ -0,0 +1,822 @@
+    ],
+  },
+];
+
+export default eslintConfig;
+`
+
+const ESLINT_CONFIG_TEMPLATE_JAVASCRIPT = `import { dirname } from "path";
+import { fileURLToPath } from "url";
+import { FlatCompat } from "@eslint/eslintrc";
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
+const compat = new FlatCompat({
+  baseDirectory: __dirname,
+});
+
+const eslintConfig = [
+  ...compat.extends("next/core-web-vitals"),
+  {
+    ignores: [
+      "node_modules/**",
+      ".next/**",
+      "out/**",
+      "build/**",
+      "next-env.d.ts",
+    ],
+  },
+];
+
+export default eslintConfig;
+`
+
+function detectTypeScript(projectRoot: string): boolean {
+  return existsSync(path.join(projectRoot, 'tsconfig.json'))
+}
+
+function findExistingEslintConfig(projectRoot: string): {
+  exists: boolean
+  path?: string
+  isFlat?: boolean
+} {
+  const flatConfigs = [
+    'eslint.config.js',
+    'eslint.config.mjs',
+    'eslint.config.cjs',
+    'eslint.config.ts',
+    'eslint.config.mts',
+    'eslint.config.cts',
+  ]
+
+  const legacyConfigs = [
+    '.eslintrc.js',
+    '.eslintrc.cjs',
+    '.eslintrc.yaml',
+    '.eslintrc.yml',
+    '.eslintrc.json',
+    '.eslintrc',
+  ]
+
+  // Check for flat configs first (preferred for v9+)
+  for (const config of flatConfigs) {
+    const configPath = path.join(projectRoot, config)
+    if (existsSync(configPath)) {
+      return { exists: true, path: configPath, isFlat: true }
+    }
+  }
+
+  // Check for legacy configs
+  for (const config of legacyConfigs) {
+    const configPath = path.join(projectRoot, config)
+    if (existsSync(configPath)) {
+      return { exists: true, path: configPath, isFlat: false }
+    }
+  }
+
+  return { exists: false }
+}
+
+function updateExistingFlatConfig(
+  configPath: string,
+  isTypeScript: boolean
+): boolean {
+  try {
+    const configContent = readFileSync(configPath, 'utf8')
+
+    // Check if Next.js configs are already imported
+    const hasNextConfigs =
+      configContent.includes('next/core-web-vitals') ||
+      configContent.includes('next/typescript')
+
+    // TypeScript config files need special handling
+    if (
+      configPath.endsWith('.ts') ||
+      configPath.endsWith('.mts') ||
+      configPath.endsWith('.cts')
+    ) {
+      console.log('   TypeScript config files require manual migration')
+      console.log('   Please add the following to your config:')
+      console.log('   - Import: import { FlatCompat } from "@eslint/eslintrc"')
+      console.log(
+        '   - Extend: ...compat.extends("next/core-web-vitals"' +
+          (isTypeScript ? ', "next/typescript"' : '') +
+          ')'
+      )
+      return false
+    }
+
+    // Parse the file using jscodeshift
+    const j = createParserFromPath(configPath)
+    const root = j(configContent)
+
+    // Determine if it's CommonJS or ES modules
+    const isCommonJS =
+      configPath.endsWith('.cjs') ||
+      (configPath.endsWith('.js') &&
```

Comment:
`configPath.endsWith('.js')` does mean it's cjs, it depends on package.json type field. if type !== "module", then `.js` extension can be treated as cjs, otherwise esm